app/src/main/java/com/theuhooi/uhooipicbook/modules/monsterlist/MonsterListFragment.kt

```diff
@@ -0,0 +1,59 @@
+package com.theuhooi.uhooipicbook.modules.monsterlist
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.theuhooi.uhooipicbook.R
+import com.theuhooi.uhooipicbook.modules.monsterlist.entity.MonsterContent
+import com.theuhooi.uhooipicbook.modules.monsterlist.entity.MonsterContent.MonsterItem
+
+class MonsterListFragment : Fragment() {
+
+    // MARK: Stored Instance Properties
+
+    private var listener: OnListFragmentInteractionListener? = null
+
+    // MARK: View Life-Cycle Methods
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_monster_list, container, false)
+        if (view is RecyclerView) {
+            view.layoutManager = GridLayoutManager(context, 1)
+            view.adapter = MonsterListRecyclerViewAdapter(MonsterContent.monsters, this.listener)
+        }
+
+        return view
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+
+        if (context is OnListFragmentInteractionListener) {
+            this.listener = context
+        } else {
+            throw RuntimeException("$context must implement OnListFragmentInteractionListener")
+        }
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+
+        this.listener = null
+    }
+
+    // MARK: Interfaces
+
+    interface OnListFragmentInteractionListener {
+        fun onListFragmentInteraction(item: MonsterItem?)
```

Comment:
`item` seems always non-null value.

Reply:
I thought so, but left it auto-generated😅
Fixed it.